schema_utils.py

```diff
@@ -104,10 +108,40 @@ def normalize_against_schema(
         assert isinstance(obj, dict), ('Expected dict, received %s' % obj)
         expected_dict_keys = [
             p[SCHEMA_KEY_NAME] for p in schema[SCHEMA_KEY_PROPERTIES]]
+
+        # For handling cases of missing args.
+        list_of_missing_args_with_default_none = []
+        for prop in schema[SCHEMA_KEY_PROPERTIES]:
+
+            if (SCHEMA_KEY_DEFAULT_VALUE not in prop[SCHEMA_KEY_SCHEMA] or
+                    prop[SCHEMA_KEY_NAME] in obj):
+                continue
+
+            default_value = prop[SCHEMA_KEY_SCHEMA][SCHEMA_KEY_DEFAULT_VALUE]
```

Comment:
Here default_value key handles both the cases (for default args as well as optional args.).
```python
schema = {
    'arg_a': {
        'type': 'int',
        'default_value': 5
    }
}
```

Reply:
> Why are we only handling default_value for dict schema? 

I think we should have a general default_value filed instead of having it only for dict properties. [Dict properties are expected to follow any other general schema of list/int/string etc.] Any specific reason for not treating "default_value" as general property?

Reply:
@DubeySandeep Let's not extend the "default value" functionality more than necessary. It actually adds branching/logic and it would be nicer not to use it in general, but in this case there's a clear use case for it being needed as part of the handler args. If, later, we need it for other types of fields, we can introduce it at that time.

Reply:
Thank you @seanlip for taking up this question.